src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,84 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try {
+            Files.deleteIfExists(Path.of(toFileName));
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot delete file " + toFileName, e);
+        }
+
+        String[] actionsList = readFile(fromFileName);
+        String report = createReport(actionsList);
+        writeReportToFile(report, toFileName);
+    }
```

Comment:
The method `getStatistic` seems to be doing multiple tasks: deleting a file, reading from a file, creating a report, and writing to a file. Consider refactoring this method to ensure that each method is responsible for one task only, in line with the checklist's guidance on dividing program logic.